**Análise de Dados**

**O primeiro passo para ánalise de dados, logicamente é a leitura do arquivo.**

In [ ]:
#O método OPEN do Python lê um arquivo e joga para uma variavel

titanic_csv = open('titanic.csv', 'r')

In [ ]:
#O método readLines identifica cada linha através do \n (Enter)
#Uma LISTA de LINHAS

titanic_csv.readlines()

In [4]:
#Algo a se ter cuidado ao Ler um arquivo no Python é que o mesmo arquivo não pode ser lido duas vezes
#isso acontece pois o cursor fica após o ultimo caracter. O método Le linhas

titanic_csv.readlines()

[]

In [5]:
#Podemos posicionar novamente o cursor na linha 0

titanic_csv.seek(0)#Retorna o arquivo para a primeira linha

0

In [6]:
#Aqui estamos lendo o arquivo e jogando o resultado em uma variavel

linhas = titanic_csv.readlines()

In [ ]:
linhas

In [8]:
#Imprimindo o primeiro elemento da Lista

linhas[0]

'PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked\n'

In [9]:
linhas[1]

'1,0,3,"Braund, Mr. Owen Harris",male,22,1,0,A/5 21171,7.25,,S\n'

In [10]:
#Note que temos o arquivo organizado em linhas e cada uma tem o seu Indice

linhas[23]

'23,1,3,"McGowan, Miss. Anna ""Annie""",female,15,0,0,330923,8.0292,,Q\n'

In [11]:
#Apos o arquivo estar na variavel, devemos fechar o arquivo para Liberar memória

titanic_csv.close() #fecha o arquivo - Retira da memoria

In [12]:
#Vamos utilizar o metodo SPLIT para separar o elemento 0 por virgulas, ou seja, separar cada cabeçalho

linhas[0].split(',')

['PassengerId',
 'Survived',
 'Pclass',
 'Name',
 'Sex',
 'Age',
 'SibSp',
 'Parch',
 'Ticket',
 'Fare',
 'Cabin',
 'Embarked\n']

In [13]:
#Podemos realizar essa operação com qualquer linha

linhas[32].split(',')

['32',
 '1',
 '1',
 '"Spencer',
 ' Mrs. William Augustus (Marie Eugenie)"',
 'female',
 '',
 '1',
 '0',
 'PC 17569',
 '146.5208',
 'B78',
 'C\n']

In [14]:
#Aqui vamos pegar a linha 0 que contem os nomes das colunas, e atribuir a uma variavel

colunas = linhas[0].split(',')

In [15]:
colunas #Agora é uma lista de strings

['PassengerId',
 'Survived',
 'Pclass',
 'Name',
 'Sex',
 'Age',
 'SibSp',
 'Parch',
 'Ticket',
 'Fare',
 'Cabin',
 'Embarked\n']

In [16]:
#Como aprendemos em Python básico, podemos acessar cada linha de uma Lista através do índice

colunas[0]

'PassengerId'

In [17]:
colunas[3]

'Name'

In [ ]:
#Cada coluna sera a chave e os valores uma lista do dicionario

In [18]:
#Vamos criar uma variavel dados, que receberá o conteudo da linha 1, separado por vírgulas, como fizemos
#com cabeçalhos

dados = linhas[1].split(',')

In [19]:
print(colunas)
print(dados)

['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked\n']
['1', '0', '3', '"Braund', ' Mr. Owen Harris"', 'male', '22', '1', '0', 'A/5 21171', '7.25', '', 'S\n']


In [20]:
#Aqui eu pego a primeira posição de cada Lista e consigo correlacionar

print(colunas[0])
print(dados[0])

PassengerId
1


In [21]:
#É uma boa prática sempre contar o numero de elementos da lista

print(len(colunas), len(dados))

12 13


In [ ]:
#Problema - o nome das pessoas tem virgula - REGEX PAL .* (qq texto qq qtde) \s -> espaço em regex

# https://www.regexpal.com/

# O obejetivo é retirar a virgula e inverter a posição dos nomes

In [22]:
import re

match = re.compile(r'"(.*)(,)(\s.*)"')
match.sub(r'"\3 \1"', linhas[1])

'1,0,3," Mr. Owen Harris Braund",male,22,1,0,A/5 21171,7.25,,S\n'

In [ ]:
#Criando uma funcao que remove a virgula nomes e troca a posicao

In [23]:
def tratar_nome(linha):
    import re
    match = re.compile(r'"(.*)(,)(\s.*)"')
    return match.sub(r'"\3 \1"', linha)

In [24]:
#Vamos testar a funcao na Linha 1

tratar_nome(linhas[1])

'1,0,3," Mr. Owen Harris Braund",male,22,1,0,A/5 21171,7.25,,S\n'

In [25]:
linha1_tratada = tratar_nome(linhas[1])

In [26]:
linha1_separada = linha1_tratada.split(",")

In [27]:
print(colunas)
print(linha1_tratada)
print(linha1_separada)

['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked\n']
1,0,3," Mr. Owen Harris Braund",male,22,1,0,A/5 21171,7.25,,S

['1', '0', '3', '" Mr. Owen Harris Braund"', 'male', '22', '1', '0', 'A/5 21171', '7.25', '', 'S\n']


In [28]:
print(len(colunas), len(linha1_separada))

12 12


In [ ]:
#Criando o dicionario - o dicionário titanic_dados está recebendo a coluna
#Como chave e uma lista vazia como valores

In [29]:
#Na criação desse dicionário, criaremos as chaves e os valores como vazios

titanic_dados = {}

for coluna in colunas:
    titanic_dados[coluna] = []
titanic_dados    

{'PassengerId': [],
 'Survived': [],
 'Pclass': [],
 'Name': [],
 'Sex': [],
 'Age': [],
 'SibSp': [],
 'Parch': [],
 'Ticket': [],
 'Fare': [],
 'Cabin': [],
 'Embarked\n': []}

In [30]:
type(titanic_dados)

dict

In [ ]:
linhas[1] #verificando cada item e cada chave

In [31]:
#Aqui trataremos todas as linhas, excluindo a linha 0, que é o cabeçalho

for dado in linhas[1:]:
    dado_tratado = tratar_nome(dado)
    print(dado_tratado)

1,0,3," Mr. Owen Harris Braund",male,22,1,0,A/5 21171,7.25,,S

2,1,1," Mrs. John Bradley (Florence Briggs Thayer) Cumings",female,38,1,0,PC 17599,71.2833,C85,C

3,1,3," Miss. Laina Heikkinen",female,26,0,0,STON/O2. 3101282,7.925,,S

4,1,1," Mrs. Jacques Heath (Lily May Peel) Futrelle",female,35,1,0,113803,53.1,C123,S

5,0,3," Mr. William Henry Allen",male,35,0,0,373450,8.05,,S

6,0,3," Mr. James Moran",male,,0,0,330877,8.4583,,Q

7,0,1," Mr. Timothy J McCarthy",male,54,0,0,17463,51.8625,E46,S

8,0,3," Master. Gosta Leonard Palsson",male,2,3,1,349909,21.075,,S

9,1,3," Mrs. Oscar W (Elisabeth Vilhelmina Berg) Johnson",female,27,0,2,347742,11.1333,,S

10,1,2," Mrs. Nicholas (Adele Achem) Nasser",female,14,1,0,237736,30.0708,,C

11,1,3," Miss. Marguerite Rut Sandstrom",female,4,1,1,PP 9549,16.7,G6,S

12,1,1," Miss. Elizabeth Bonnell",female,58,0,0,113783,26.55,C103,S

13,0,3," Mr. William Henry Saundercock",male,20,0,0,A/5. 2151,8.05,,S

14,0,3," Mr. Anders Johan Andersson",male,39,1,5,34

In [32]:
#Aqui faremos a mesma coisa, porém vamos separar as linhas pela virgula e converter em uma lista

for dado in linhas[1:]:
    dado_tratado = tratar_nome(dado)
    dado_como_lista = dado_tratado.split(',')
    print(dado_como_lista)
    

['1', '0', '3', '" Mr. Owen Harris Braund"', 'male', '22', '1', '0', 'A/5 21171', '7.25', '', 'S\n']
['2', '1', '1', '" Mrs. John Bradley (Florence Briggs Thayer) Cumings"', 'female', '38', '1', '0', 'PC 17599', '71.2833', 'C85', 'C\n']
['3', '1', '3', '" Miss. Laina Heikkinen"', 'female', '26', '0', '0', 'STON/O2. 3101282', '7.925', '', 'S\n']
['4', '1', '1', '" Mrs. Jacques Heath (Lily May Peel) Futrelle"', 'female', '35', '1', '0', '113803', '53.1', 'C123', 'S\n']
['5', '0', '3', '" Mr. William Henry Allen"', 'male', '35', '0', '0', '373450', '8.05', '', 'S\n']
['6', '0', '3', '" Mr. James Moran"', 'male', '', '0', '0', '330877', '8.4583', '', 'Q\n']
['7', '0', '1', '" Mr. Timothy J McCarthy"', 'male', '54', '0', '0', '17463', '51.8625', 'E46', 'S\n']
['8', '0', '3', '" Master. Gosta Leonard Palsson"', 'male', '2', '3', '1', '349909', '21.075', '', 'S\n']
['9', '1', '3', '" Mrs. Oscar W (Elisabeth Vilhelmina Berg) Johnson"', 'female', '27', '0', '2', '347742', '11.1333', '', 'S\n']


In [33]:
#Aqui faremos a mesma coisa dos passos anteriores, porem vamos criar uma lista para
#adicionar cada linha convertida em lista, fazendo uma lista aninhada

dados_tratados = []

for dado in linhas[1:]:
    dado_tratado = tratar_nome(dado)
    dado_como_lista = dado_tratado.split(',')
    dados_tratados.append(dado_como_lista)

In [34]:
print(dados_tratados[5][3])

" Mr. James Moran"


In [35]:
#Coluna 1 e 4 do Cabeçalho. Colunas 1e 4 da lisnha 1. Conseguimos correlacionar!

print(colunas[1], colunas[3])
print(dados_tratados[34][1], dados_tratados[34][3])

Survived Name
0 " Mr. Edgar Joseph Meyer"


In [42]:
#Note que colunas é a nossa lista de cabeçalhos separada, e passageiro é a variavel que estamos 
#criando para o FOR, e vai percorrer dados_tratados

for passageiro in dados_tratados:
    print(colunas[3], passageiro[3], colunas[1], passageiro[1],  colunas[2], passageiro[2]) 
    
    #Mesmo indice colunas e o passageiro

Name " Mr. Owen Harris Braund" Survived 0 Pclass 3
Name " Mrs. John Bradley (Florence Briggs Thayer) Cumings" Survived 1 Pclass 1
Name " Miss. Laina Heikkinen" Survived 1 Pclass 3
Name " Mrs. Jacques Heath (Lily May Peel) Futrelle" Survived 1 Pclass 1
Name " Mr. William Henry Allen" Survived 0 Pclass 3
Name " Mr. James Moran" Survived 0 Pclass 3
Name " Mr. Timothy J McCarthy" Survived 0 Pclass 1
Name " Master. Gosta Leonard Palsson" Survived 0 Pclass 3
Name " Mrs. Oscar W (Elisabeth Vilhelmina Berg) Johnson" Survived 1 Pclass 3
Name " Mrs. Nicholas (Adele Achem) Nasser" Survived 1 Pclass 2
Name " Miss. Marguerite Rut Sandstrom" Survived 1 Pclass 3
Name " Miss. Elizabeth Bonnell" Survived 1 Pclass 1
Name " Mr. William Henry Saundercock" Survived 0 Pclass 3
Name " Mr. Anders Johan Andersson" Survived 0 Pclass 3
Name " Miss. Hulda Amanda Adolfina Vestrom" Survived 0 Pclass 3
Name " Mrs. (Mary D Kingcome)  Hewlett" Survived 1 Pclass 2
Name " Master. Eugene Rice" Survived 0 Pclass 3
Name " 

# Utilizando o ENUMERATE. O enumerate retorna o Índice em que o elemento está e tambem o seu valor!

In [39]:
frutas = ['mamão', 'melão', 'melancia']

for indice, fruta in enumerate(frutas):
    print('Fruta:', fruta, 'no indice', indice)

Fruta: mamão no indice 0
Fruta: melão no indice 1
Fruta: melancia no indice 2


In [44]:
print(colunas)
print('Coluna no indice 10', colunas[10])

['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked\n']
Coluna no indice Cabin


In [45]:
#Pegando o indice, posso correlacionar com as colunas

for passageiro in dados_tratados:
    print(passageiro)
    for indice, dado in enumerate(passageiro):
        print('Indice',indice,'DADO:',dado)

['1', '0', '3', '" Mr. Owen Harris Braund"', 'male', '22', '1', '0', 'A/5 21171', '7.25', '', 'S\n']
Indice 0 DADO: 1
Indice 1 DADO: 0
Indice 2 DADO: 3
Indice 3 DADO: " Mr. Owen Harris Braund"
Indice 4 DADO: male
Indice 5 DADO: 22
Indice 6 DADO: 1
Indice 7 DADO: 0
Indice 8 DADO: A/5 21171
Indice 9 DADO: 7.25
Indice 10 DADO: 
Indice 11 DADO: S

['2', '1', '1', '" Mrs. John Bradley (Florence Briggs Thayer) Cumings"', 'female', '38', '1', '0', 'PC 17599', '71.2833', 'C85', 'C\n']
Indice 0 DADO: 2
Indice 1 DADO: 1
Indice 2 DADO: 1
Indice 3 DADO: " Mrs. John Bradley (Florence Briggs Thayer) Cumings"
Indice 4 DADO: female
Indice 5 DADO: 38
Indice 6 DADO: 1
Indice 7 DADO: 0
Indice 8 DADO: PC 17599
Indice 9 DADO: 71.2833
Indice 10 DADO: C85
Indice 11 DADO: C

['3', '1', '3', '" Miss. Laina Heikkinen"', 'female', '26', '0', '0', 'STON/O2. 3101282', '7.925', '', 'S\n']
Indice 0 DADO: 3
Indice 1 DADO: 1
Indice 2 DADO: 3
Indice 3 DADO: " Miss. Laina Heikkinen"
Indice 4 DADO: female
Indice 5 DADO: 

Indice 8 DADO: 7534
Indice 9 DADO: 9.8458
Indice 10 DADO: 
Indice 11 DADO: S

['878', '0', '3', '" Mr. Nedelio Petroff"', 'male', '19', '0', '0', '349212', '7.8958', '', 'S\n']
Indice 0 DADO: 878
Indice 1 DADO: 0
Indice 2 DADO: 3
Indice 3 DADO: " Mr. Nedelio Petroff"
Indice 4 DADO: male
Indice 5 DADO: 19
Indice 6 DADO: 0
Indice 7 DADO: 0
Indice 8 DADO: 349212
Indice 9 DADO: 7.8958
Indice 10 DADO: 
Indice 11 DADO: S

['879', '0', '3', '" Mr. Kristo Laleff"', 'male', '', '0', '0', '349217', '7.8958', '', 'S\n']
Indice 0 DADO: 879
Indice 1 DADO: 0
Indice 2 DADO: 3
Indice 3 DADO: " Mr. Kristo Laleff"
Indice 4 DADO: male
Indice 5 DADO: 
Indice 6 DADO: 0
Indice 7 DADO: 0
Indice 8 DADO: 349217
Indice 9 DADO: 7.8958
Indice 10 DADO: 
Indice 11 DADO: S

['880', '1', '1', '" Mrs. Thomas Jr (Lily Alexenia Wilson) Potter"', 'female', '56', '0', '1', '11767', '83.1583', 'C50', 'C\n']
Indice 0 DADO: 880
Indice 1 DADO: 1
Indice 2 DADO: 1
Indice 3 DADO: " Mrs. Thomas Jr (Lily Alexenia Wilson) Potter"
I

In [46]:
# 1 - Pego cada linha da variavel dados_tratados e coloco em passageiro
# 2 - Faço um for em passageiro e pego o indice e o restante dos dados
# 3 - Vou na variavel coluna e pego a coluna no respectivo indice

for passageiro in dados_tratados:
    for indice, dado in enumerate(passageiro): #Enumerate retorna o INDICE e VALOR(atribuição MULTIPLA)
        coluna = colunas[indice]
        print(coluna,dado)


PassengerId 1
Survived 0
Pclass 3
Name " Mr. Owen Harris Braund"
Sex male
Age 22
SibSp 1
Parch 0
Ticket A/5 21171
Fare 7.25
Cabin 
Embarked
 S

PassengerId 2
Survived 1
Pclass 1
Name " Mrs. John Bradley (Florence Briggs Thayer) Cumings"
Sex female
Age 38
SibSp 1
Parch 0
Ticket PC 17599
Fare 71.2833
Cabin C85
Embarked
 C

PassengerId 3
Survived 1
Pclass 3
Name " Miss. Laina Heikkinen"
Sex female
Age 26
SibSp 0
Parch 0
Ticket STON/O2. 3101282
Fare 7.925
Cabin 
Embarked
 S

PassengerId 4
Survived 1
Pclass 1
Name " Mrs. Jacques Heath (Lily May Peel) Futrelle"
Sex female
Age 35
SibSp 1
Parch 0
Ticket 113803
Fare 53.1
Cabin C123
Embarked
 S

PassengerId 5
Survived 0
Pclass 3
Name " Mr. William Henry Allen"
Sex male
Age 35
SibSp 0
Parch 0
Ticket 373450
Fare 8.05
Cabin 
Embarked
 S

PassengerId 6
Survived 0
Pclass 3
Name " Mr. James Moran"
Sex male
Age 
SibSp 0
Parch 0
Ticket 330877
Fare 8.4583
Cabin 
Embarked
 Q

PassengerId 7
Survived 0
Pclass 1
Name " Mr. Timothy J McCarthy"
Sex male
Age 54

# linhas = todo o arquivo, sem nenhum tratamento, onde cada indice é uma linha, por exemplo, linhas[0]

# colunas = ['Passengerid', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSP', 'Parch', 'Ticket', 'Fare'...]

# dados = ['1','0','3',"'Braund', 'Mr. Owen Harris'", 'male', '22', '1', '0', 'A/5 21171', '7.25', ", 'S\n']

# titanic_dados = O dicionario com as chaves (colunas) sem os valores

In [47]:
print(titanic_dados)

{'PassengerId': [], 'Survived': [], 'Pclass': [], 'Name': [], 'Sex': [], 'Age': [], 'SibSp': [], 'Parch': [], 'Ticket': [], 'Fare': [], 'Cabin': [], 'Embarked\n': []}


In [48]:
for passageiro in dados_tratados:
    for indice, dado in enumerate(passageiro):
        coluna = colunas[indice]
        titanic_dados[coluna].append(dado)

In [49]:
titanic_dados

{'PassengerId': ['1',
  '2',
  '3',
  '4',
  '5',
  '6',
  '7',
  '8',
  '9',
  '10',
  '11',
  '12',
  '13',
  '14',
  '15',
  '16',
  '17',
  '18',
  '19',
  '20',
  '21',
  '22',
  '23',
  '24',
  '25',
  '26',
  '27',
  '28',
  '29',
  '30',
  '31',
  '32',
  '33',
  '34',
  '35',
  '36',
  '37',
  '38',
  '39',
  '40',
  '41',
  '42',
  '43',
  '44',
  '45',
  '46',
  '47',
  '48',
  '49',
  '50',
  '51',
  '52',
  '53',
  '54',
  '55',
  '56',
  '57',
  '58',
  '59',
  '60',
  '61',
  '62',
  '63',
  '64',
  '65',
  '66',
  '67',
  '68',
  '69',
  '70',
  '71',
  '72',
  '73',
  '74',
  '75',
  '76',
  '77',
  '78',
  '79',
  '80',
  '81',
  '82',
  '83',
  '84',
  '85',
  '86',
  '87',
  '88',
  '89',
  '90',
  '91',
  '92',
  '93',
  '94',
  '95',
  '96',
  '97',
  '98',
  '99',
  '100',
  '101',
  '102',
  '103',
  '104',
  '105',
  '106',
  '107',
  '108',
  '109',
  '110',
  '111',
  '112',
  '113',
  '114',
  '115',
  '116',
  '117',
  '118',
  '119',
  '120',
  '121',
  '1

In [50]:
int(titanic_dados["Age"][0])

22

In [51]:
for dado in titanic_dados["Age"]:
    print(dado)

22
38
26
35
35

54
2
27
14
4
58
20
39
14
55
2

31

35
34
15
28
8
38

19


40


66
28
42

21
18
14
40
27

3
19




18
7
21
49
29
65

21
28.5
5
11
22
38
45
4


29
19
17
26
32
16
21
26
32
25


0.83
30
22
29

28
17
33
16

23
24
29
20
46
26
59

71
23
34
34
28

21
33
37
28
21

38

47
14.5
22
20
17
21
70.5
29
24
2
21

32.5
32.5
54
12

24

45
33
20
47
29
25
23
19
37
16
24

22
24
19
18
19
27
9
36.5
42
51
22
55.5
40.5

51
16
30


44
40
26
17
1
9

45

28
61
4
1
21
56
18

50
30
36


9
1
4


45
40
36
32
19
19
3
44
58

42

24
28

34
45.5
18
2
32
26
16
40
24
35
22
30

31
27
42
32
30
16
27
51

38
22
19
20.5
18

35
29
59
5
24

44
8
19
33


29
22
30
44
25
24
37
54

29
62
30
41
29

30
35
50

3
52
40

36
16
25
58
35

25
41
37

63
45

7
35
65
28
16
19

33
30
22
42
22
26
19
36
24
24

23.5
2

50


19


0.92

17
30
30
24
18
26
28
43
26
24
54
31
40
22
27
30
22

36
61
36
31
16

45.5
38
16


29
41
45
45
2
24
28
25
36
24
40

3
42
23

15
25

28
22
38


40
29
45
35

30
60


24
25
18
19
22
3

22
27
20
19
42
1
32
35


In [52]:
for indice, age in enumerate(titanic_dados["Age"]):
    if titanic_dados['Age'][indice] != '':
        titanic_dados['Age'][indice] = float(age)
    else:
        titanic_dados['Age'][indice] = -1

In [53]:
type(titanic_dados['Age'][0])

float

In [54]:
print(max(titanic_dados["Age"]))

80.0


In [55]:
print(min(titanic_dados["Age"]))

-1


In [56]:
lista_idades = []

for idade in titanic_dados["Age"]:
    if idade != -1:
        lista_idades.append(idade)


In [57]:
def media(lista):
    soma_total = sum(lista)
    quantidade = len(lista)
    return soma_total / quantidade
    

In [58]:
print(media(lista_idades))

29.69911764705882


In [59]:
media_idades = media(lista_idades)

In [60]:
for indice, age in enumerate(titanic_dados['Age']):
    if age == 1:
        titanic_dados['Age'][indice] = media_idades

In [64]:
import pandas
titanic_dados = pandas.read_csv('titanic.csv')
titanic_dados.head(50)

PassengerId  Survived  Pclass  \
0             1         0       3   
1             2         1       1   
2             3         1       3   
3             4         1       1   
4             5         0       3   
5             6         0       3   
6             7         0       1   
7             8         0       3   
8             9         1       3   
9            10         1       2   
10           11         1       3   
11           12         1       1   
12           13         0       3   
13           14         0       3   
14           15         0       3   
15           16         1       2   
16           17         0       3   
17           18         1       2   
18           19         0       3   
19           20         1       3   
20           21         0       2   
21           22         1       2   
22           23         1       3   
23           24         1       1   
24           25         0       3   
25           26         1       3   
26           27         0       3   
27           28         0       1   
28           29         1       3   
29           30         0       3   
30           31         0       1   
31           32         1       1   
32           33         1       3   
33           34         0       2   
34           35         0       1   
35           36         0       1   
36           37         1       3   
37           38         0       3   
38           39         0       3   
39           40         1       3   
40           41         0       3   
41           42         0       2   
42           43         0       3   
43           44         1       2   
44           45         1       3   
45           46         0       3   
46           47         0       3   
47           48         1       3   
48           49         0       3   
49           50         0       3   

                                                 Name     Sex   Age  SibSp  \
0                             Braund, Mr. Owen Harris    male  22.0      1   
1   Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                              Heikkinen, Miss. Laina  female  26.0      0   
3        Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                            Allen, Mr. William Henry    male  35.0      0   
5                                    Moran, Mr. James    male   NaN      0   
6                             McCarthy, Mr. Timothy J    male  54.0      0   
7                      Palsson, Master. Gosta Leonard    male   2.0      3   
8   Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                 Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   
10                    Sandstrom, Miss. Marguerite Rut  female   4.0      1   
11                           Bonnell, Miss. Elizabeth  female  58.0      0   
12                     Saundercock, Mr. William Henry    male  20.0      0   
13                        Andersson, Mr. Anders Johan    male  39.0      1   
14               Vestrom, Miss. Hulda Amanda Adolfina  female  14.0      0   
15                   Hewlett, Mrs. (Mary D Kingcome)   female  55.0      0   
16                               Rice, Master. Eugene    male   2.0      4   
17                       Williams, Mr. Charles Eugene    male   NaN      0   
18  Vander Planke, Mrs. Julius (Emelia Maria Vande...  female  31.0      1   
19                            Masselmani, Mrs. Fatima  female   NaN      0   
20                               Fynney, Mr. Joseph J    male  35.0      0   
21                              Beesley, Mr. Lawrence    male  34.0      0   
22                        McGowan, Miss. Anna "Annie"  female  15.0      0   
23                       Sloper, Mr. William Thompson    male  28.0      0   
24                      Palsson, Miss. Torborg Danira  female   8.0      3   
25  Asplund, Mrs. Carl Oscar (Selma Augusta Emilia...  female  38.0      1   
26        